bugwarrior/services/gmail.py

```diff
@@ -90,20 +91,18 @@ def __init__(self, *args, **kw):
 
         self.query = self.config.get('query', 'label:Starred')
         self.login_name = self.config.get('login_name', 'me')
-        self.client_secret_path = self.get_config_path('client_secret_path')
-        self.credentials_path = self.get_config_path('credentials_path')
+        self.client_secret_path = self.get_config_path(
+                'client_secret_path',
+                self.DEFAULT_CLIENT_SECRET_PATH)
+        credentials_name = clean_filename(self.login_name
+                if self.login_name != 'me' else self.target)
+        self.credentials_path = os.path.join(
+                self.config.data.path,
+                'gmail_credentials_%s.json' % (credentials_name,))
         self.gmail_api = self.build_api()
 
-    @classmethod
-    def validate_config(cls, service_config, target):
-        for option in ('client_secret_path', 'credentials_path'):
-            if option not in service_config:
-                die("[%s] has no '%s.%s'" % (target, cls.CONFIG_PREFIX, option))
-
-        IssueService.validate_config(service_config, target)
```

Comment:
Yes, there's no longer any validation to do (all parameters are optional). So I'm happy to leave this to ``IssueService``'s implementation.